### PR TITLE
resource_manager: Remove `buffer` and `initialRequestUnits` var

### LINKED ISSUE #6185: Acquire tokens abnormally after stopping workload And Token State is unexpected
### What did you expect to see?
After stopping workload, report consumption by report-period.
After stopping workload, the token should be 500 in server.

### What did you see instead?
1.
 [2023/03/16 12:58:29.228 +00:00] [DEBUG] [controller.go:342] ["[resource group controller] send token bucket request"] [keyspaceName=2XCiA4qTMHwJZJC] [keyspaceID=5056] [now=2023/03/16 12:58:29.228 +00:00] [req="[{\"resource_group_name\":\"5056\",\"Request\":{\"ru_items\":{\"request_r_u\":[{\"value\":1504.6852965871174}]}},\"consumption_since_last_request\":{\"r_r_u\":3.51568603515625,\"read_bytes\":1028,\"total_cpu_time_ms\":90,\"sql_layer_cpu_time_ms\":90,\"kv_read_rpc_count\":14}}]"] [source=low_ru]

but avg ru speed is 1.1
2.
image

### LINKED ISSUE #6187: Exceeded resource group quota limitation when the workload of the same resouce group ran on multiple TiDB
### What did you do?
Deploy a cluster with 2 TiDB and set a resource group named `rg1`.
Run two sysbench processes on 2 TiDB for specific resource group rg1.

### What did you expect to see?
After running cmd on TiDB1, can also run sysbench on TiDB2

### What did you see instead?
image

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: Close #6185 Close #6187

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/resource_group/controller/controller.go [hunk 1]
```diff
@@ -723,6 +731,9 @@ func (gc *groupCostController) calcAvg(counter *tokenCounter, new float64) bool
 
 func (gc *groupCostController) shouldReportConsumption() bool {
 	timeSinceLastRequest := gc.run.now.Sub(gc.run.lastRequestTime)
+	if !gc.run.initialRequestCompleted {
+		return true
+	}
```
Comment (HuSharp) [inline client/resource_group/controller/controller.go:736]: can move this if block to the top

**Comment by nolouch** [comment]
test failed.

File: client/resource_group/controller/controller.go [hunk 2]
```diff
@@ -958,8 +954,12 @@ func (gc *groupCostController) collectRequestAndConsumption(selectTyp selectType
...
 
 func (gc *groupCostController) calcRequest(counter *tokenCounter) float64 {
-	value := counter.avgRUPerSec*gc.run.targetPeriod.Seconds() + bufferRUs
-	value -= counter.limiter.AvailableTokens(gc.run.now)
+	value := counter.avgRUPerSec * gc.run.targetPeriod.Seconds() * needTokensAmplification
+	if gc.run.initialRequestCompleted {
+		value -= counter.limiter.AvailableTokens(gc.run.now)
+	} else {
+		value += counter.initialToken - counter.limiter.AvailableTokens(gc.run.now)
+	}
```
Comment (HuSharp) [inline client/resource_group/controller/controller.go:962]: Perhaps `value -= counter.limiter.AvailableTokens(gc.run.now)' can be extracted out.

File: client/resource_group/controller/controller.go [hunk 3]
```diff
@@ -556,52 +559,60 @@ func newGroupCostController(
 func (gc *groupCostController) initRunState() {
 	now := time.Now()
 	gc.run.now = now
-	gc.run.lastRequestTime = now
+	gc.run.lastRequestTime = now.Add(-defaultTargetPeriod)
```
Comment (HuSharp) [inline client/resource_group/controller/controller.go:562]: Why not sub?

File: client/resource_group/controller/controller.go [hunk 4]
```diff
@@ -958,8 +954,12 @@ func (gc *groupCostController) collectRequestAndConsumption(selectTyp selectType
 }
 
 func (gc *groupCostController) calcRequest(counter *tokenCounter) float64 {
-	value := counter.avgRUPerSec*gc.run.targetPeriod.Seconds() + bufferRUs
-	value -= counter.limiter.AvailableTokens(gc.run.now)
+	value := counter.avgRUPerSec * gc.run.targetPeriod.Seconds() * needTokensAmplification
```
Comment (HuSharp) [inline client/resource_group/controller/controller.go:957]: Maybe can add some comments for `needTokensAmplification`

File: client/resource_group/controller/controller.go (hunk 3 above)
Comment (CabinfeverB) [reply client/resource_group/controller/controller.go:562]: `Sub` is used to get the duration between two `Time`s

File: client/resource_group/controller/controller.go [hunk 5]
```diff
@@ -188,6 +189,10 @@ func (c *ResourceGroupsController) Start(ctx context.Context) {
 		failpoint.Inject("fastCleanup", func() {
 			cleanupTicker.Stop()
 			cleanupTicker = time.NewTicker(100 * time.Millisecond)
+			// because of checking `gc.run.consumption` in cleanupTicker,
+			// so should also change the stateUpdateTicker.
+			stateUpdateTicker.Stop()
+			stateUpdateTicker = time.NewTicker(200 * time.Millisecond)
```
Comment (nolouch) [inline client/resource_group/controller/controller.go:195]: why not use `defaultGroupStateUpdateInterval`?

**Review by nolouch — APPROVED** [review]

**Comment by nolouch** [comment]
PTAL @HuSharp @JmPotato

**Review by JmPotato — APPROVED** [review]

**Review by HuSharp — APPROVED** [review]
LGTM

**Pull request merged**